kernel/src/path.rs

```diff
@@ -394,6 +411,20 @@ mod tests {
         assert!(!log_path.is_commit());
         assert!(!log_path.is_checkpoint());
         assert!(log_path.is_unknown());
+
+        // Boundary test - UUID with exactly 35 characters (one too short)
+        let log_path = table_log_dir
+            .join("00000000000000000010.checkpoint.3a0d65cd-4056-49b8-937b-95f9e3ee90e.parquet")
+            .unwrap();
+        let result = ParsedLogPath::try_from(log_path);
+        assert!(
+            result.is_err(),
+            "Expected an error for UUID with exactly 35 characters"
+        );
+        assert!(
+            matches!(result, Err(Error::InvalidLogPath(_))),
+            "Expected InvalidLogPath error for boundary UUID length"
+        );
```

Comment:
I think the second assert fully covers the first?
```suggestion
        assert!(
            matches!(result, Err(Error::InvalidLogPath(_))),
            "Expected an error for UUID with exactly 35 characters"
        );
```